src/components/deck/index.js

```diff
@@ -79,8 +88,20 @@ const Deck = ({
   ...rest
 }) => {
   // Check for slides and then number slides.
+
   const filteredChildren = Array.isArray(children)
-    ? children.filter(x => isComponentType(x, 'Slide'))
+    ? children
+        .map(x => {
+          if (
+            isComponentType(x, 'Markdown') &&
+            Boolean(x.props.containsSlides)
+          ) {
+            return splitMarkdownIntoSlides(x.props.children);
+          }
+          return x;
+        })
+        .flat(1)
+        .filter(x => isComponentType(x, 'Slide'))
```

Comment:
The splitting of Markdown Slides into Slide components must occur in the Deck because that's where the logic behind how many slides and transitions occur.